smarts/core/scenario.py

```diff
@@ -559,10 +567,10 @@ def discover_missions_of_traffic_histories(self) -> Dict[str, Mission]:
             pos_x, pos_y, heading, speed = pphs
             entry_tactic = default_entry_tactic(speed)
             v_id = str(row[0])
+            veh_type = self._traffic_history.vehicle_type(v_id)
+            veh_length, veh_width, veh_height = self._traffic_history.vehicle_size(v_id)
             # missions start from front bumper, but pos is center of vehicle
-            # vehicle_length, _ = self._traffic_history.vehicle_size(v_id)
-            vehicle_length = 3.68  # TODO: agent vehicles need to be resizable!
-            hhx, hhy = radians_to_vec(heading) * (0.5 * vehicle_length)
+            hhx, hhy = radians_to_vec(heading) * (0.5 * veh_length)
             vehicle_missions[v_id] = Mission(
                 start=Start(
                     (pos_x + map_offset[0] + hhx, pos_y + map_offset[1] + hhy),
```

Comment:
Note: as I commented on earlier, the type `Start` should probably be replaced with `Pose` for the initial position in Missions, which would clean up the above a bit.  But I'm trying to avoid that sort of snowballing right now...